LastHW)/Write_article.py

```diff
@@ -0,0 +1,246 @@
+            except:
+                print('Incorrect date format')
+                expiration_date = list(datefinder.find_dates(
+                    input('Till what date we should keep it?\nPlease follow DD-MM-YYYY date version:')))[0].date()
+            return expiration_date
+        elif header == 'Consider':
+            score = random.randint(1, 100)
+            return score
+
+    def get_file_choise(self):
+        versions = ['json', 'txt', 'xml']
+        while True:
+            file_choise = input('Choose file type you want to use? Answer: json/txt/xml\n:')
+            if file_choise not in versions:
+                print('Please select correct file type')
+            else:
+                break
+        return file_choise
+
+    def check_path(self, header, file_choise):
+        user_input = input("Enter the path of your file: ")
+        if os.path.exists(user_input):
+            try:
+                file_to_parse = open(user_input, 'r+')
+                return file_to_parse
+            except:
+                print("File is not valid")
+        else:
+            print("I'll use default file ")
+            config = ConfigParser()
+            config.read('config.ini')
+            file_to_parse = open((config.get('Defaults', f'default_file_{header}_{file_choise}')), 'r+')
+            return file_to_parse
+
+    def parse_file(self, file_choise, file_to_parse):
+        text_dict = {}
+        if file_choise == 'json':
+            parsed_text_json = json.load(file_to_parse)
+            for i in parsed_text_json:
+                text_dict[i["text"]] = i["attrib"]
+        elif file_choise == 'txt':
+            parsed_text_txt = file_to_parse.read().split('\n\n')
+            for i in parsed_text_txt:
+                real_text = i.split('attrib ')
+                text_dict[real_text[0]] = real_text[1]
+            file_to_parse.close()
+        elif file_choise == 'xml':
+            tree = ET.parse(file_to_parse)
+            root = tree.getroot()
+            for elem in root:
+                text_dict[elem.text] = elem.attrib['attrib']
+        return text_dict
+
+
+class ConnectDatabase:
+    def __init__(self):
+        with  sqlite3.connect('database.db') as self.conn:
+            self.cursor = self.conn.cursor()
+            self.cursor.execute('create table if not exists news (text_news text, location text, UNIQUE (text_news, location))')
+            self.cursor.execute('create table if not exists advs (text_advs text, expiration_date date, UNIQUE (text_advs, expiration_date))')
+            self.cursor.execute('create table if not exists consider (text_consider text, score int, UNIQUE (text_consider, score))')
+
+    def insert(self, table_name,  column, column1, text, attrib):
+        with self.conn:
+            self.cursor.execute(f'insert or ignore into {table_name} ({column}, {column1}) values ( ?, ?)', (text, attrib))
```

Comment:
You use the **execute** command here. Just for your info: batch insert with **executemany** will be more efficient, and the performance difference will typically be quite large as your number of records gets large

Reply:
For example:
```
import sqlite3
con = sqlite3.connect("samples.db")
cur = con.cursor()

examples = [(2, "def"), (3, "ghi"), (4, "jkl")]
cur.executemany("INSERT INTO samples VALUES (?, ?)", examples)

for row in cur.execute("SELECT * FROM samples"):
    print(row)
```